packages/block-editor/src/components/block-patterns-list/index.js

```diff
@@ -137,7 +137,12 @@ function BlockPatternList( {
 				const isShown = shownPatterns.includes( pattern );
 				return isShown ? (
 					<BlockPattern
-						key={ pattern.name }
+						key={
+							// User added unsynced patterns do not have a unique name so we use the id instead.
+							pattern.name === 'core/block'
+								? pattern.id
+								: pattern.name
+						}
```

Comment:
Nit: Would it be worth extracting this out rather than repeating the same code below? It would also mean a single comment covers both.

Reply:
Good idea, done.